### DISCUSSION TITLE
Generate `typed-router.d.ts` when build

### DISCUSSION THREAD

**Discussion by manmen-mi** [posva/unplugin-vue-router#140] [Ideas]
Thanks for works,
I've got some error and feature request

###### sorry for machine translated english below

If you ignore the `typed-router.d.ts` file in Git, `build` command will fail.
(`error TS2307: Cannot find module 'vue-router/auto' or its corresponding type declarations.`)

If the file is managed with git, `typed-router.d.ts` conflicts too many time, so it is better to create it when executing the build.

**Comment by posva** [comment]
It should already be created. Are you running the typecheck before running the build? Or maybe are you running a build process that somehow ignores the generation of the d.ts?

**Reply by manmen-mi** [reply]
thanks for response.

> It should already be created. Are you running the typecheck before running the build?

I've checked `typed-router.d.ts` is generated when running `vite build`, but my build command is `vue-tsc --noEmit && vite build` (maybe vite-project default), It looks typechecking before build.

while i include `typed-router.d.ts` in `.gitignore`, the build will still fail.
Is there some command to generate `typed-router.d.ts`?
It would be great if I could execute like `unplugin-vue-router generate && vue-tsc --noEmit && vite build`.

as you know, The easiest way is to exclude `typed-router.d.ts` in `.gitignore`, but it would be nice to have other options like above.

**Reply by posva** [reply]
Run the `vite build` before `vue-tsc` and it should work 👍 
As unplugin-vue-router might rely on the bundler process to gather information, it's not really viable to create a generator command
